packages/ui/src/components/vuestic-components/va-datepicker/components/VaMonthPicker/VaMonthPicker.vue

```diff
@@ -0,0 +1,228 @@
+
+    const updateModelValue = (date: Date) => {
+      if (isSingleDate(modelValue.value)) {
+        emit('update:modelValue', date)
+      } else if (isPeriod(modelValue.value)) {
+        if (isDatesEqual(modelValue.value.start, date) || isDatesEqual(modelValue.value.end, date)) { return }
+
+        if (modelValue.value.end !== null) {
+          emit('update:modelValue', { start: date, end: null })
+          return
+        }
+
+        if (date < modelValue.value.start) {
+          emit('update:modelValue', { start: date, end: modelValue.value.start })
+        } else {
+          emit('update:modelValue', { start: modelValue.value.start, end: date })
+        }
+      } else if (isDates(modelValue.value)) {
+        if (isDatesArrayInclude(modelValue.value, date)) {
+          emit('update:modelValue', modelValue.value.filter((d) => !isDatesEqual(d, date)))
+        } else {
+          emit('update:modelValue', [...modelValue.value, date].sort((a, b) => a.getTime() - b.getTime()))
+        }
+      }
+    }
+
+    const onMonthClick = (year: number, month: number) => {
+      const date = new Date(year, month)
+
+      if (shouldUpdateModelValue.value) {
+        updateModelValue(date)
+      }
+
+      emit('click:month', { year, month, date })
+    }
+
+    const compareMonths = (date1: Date, date2: Date) => date1.getMonth() === date2.getMonth() && date1.getFullYear() === date2.getFullYear()
+
+    const isMonthCurrentValue = (year: number, month: number) => {
+      const date = new Date(year, month)
+
+      if (isSingleDate(modelValue.value)) {
+        return compareMonths(modelValue.value, date)
+      } else if (isDates(modelValue.value)) {
+        return modelValue.value.find((d) => compareMonths(d, date))
+      } else if (isPeriod(modelValue.value)) {
+        return isDatesEqual(modelValue.value.start, date) || isDatesEqual(modelValue.value.end, date)
+      }
+    }
+
+    const isMonthInRange = (year: number, month: number) => {
+      const date = new Date(year, month)
+
+      if (!isPeriod(modelValue.value)) { return }
+
+      if (modelValue.value.end === null) {
+        return modelValue.value.start < date
+          ? (hoveredMonth.value && hoveredMonth.value >= date)
+          : (hoveredMonth.value && hoveredMonth.value <= date)
+      }
+
+      return modelValue.value.start < date && modelValue.value.end > date
+    }
+
+    return {
+      months,
+      hoveredMonth,
+      onMonthClick,
+      isMonthCurrentValue,
+      isMonthInRange,
+    }
+  },
+})
+</script>
+
+<style lang="scss">
+$cell-height: 34px;
+
+.va-date-picker-month-calendar {
+  &__picker {
+    .weekmonths {
+      display: flex;
+
+      &__weekmonth-cell {
+        width: calc(100% / 7);
+        text-align: center;
+        font-size: 9px;
+        color: var(--va-secondary);
+        font-weight: bold;
+        height: $cell-height;
+        line-height: $cell-height;
+      }
+    }
+  }
+
+  .calendar {
```

Comment:
Avoid using simple class names to avoid conflicts.

Our core feature is ability to use in context of existing app.